### PR TITLE
insights: change series_id to string (hash); add index

### PR DESCRIPTION
**Pull request by slimsag**
Today, `series_id` will need to be some unique identifier of the actual series definition in the user/org/global settings, e.g. there are two unique series here:

```
"insights": [
    {
      "title": "fmt usage",
      "description": "fmt.Errorf/fmt.Printf usage",
      "series": [
        {
          "label": "fmt.Errorf",
          "search": "errorf",
        },
        {
          "label": "printf",
          "search": "fmt.Printf",
        }
      ]
    }
  ]
```

The plan was to have `series_id` effectively be a hash of the source of settings (user ID / org ID / global) + the actual info associated with the insight series. However, an int32 is likely not sufficient for this due to collisions. I considered using crc64 with an int64 data type, but there is still some uncomfortable room for collisions there and text is cheap - so let's just store text for now and use a sha256 hash to err on the side of caution.

### CODE REVIEW

File: migrations/frontend/1528395784_insights_series_id.down.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+BEGIN;
+
```
Comment (efritz) [inline migrations/frontend/1528395784_insights_series_id.down.sql:2]: No down migration for this?

File: migrations/frontend/1528395784_insights_series_id.up.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+ALTER TABLE insights_query_runner_jobs ALTER COLUMN series_id TYPE text;
```
Comment (efritz) [inline migrations/frontend/1528395784_insights_series_id.up.sql:3]: Title of the PR seems off.

**Review by efritz — APPROVED** [review]

File: migrations/frontend/1528395784_insights_series_id.down.sql (hunk 1 above)
Comment (slimsag) [reply migrations/frontend/1528395784_insights_series_id.down.sql:2]: I'm not sure it's useful? What would it do, change the string back to an integer?

File: migrations/frontend/1528395784_insights_series_id.up.sql (hunk 2 above)
Comment (slimsag) [reply migrations/frontend/1528395784_insights_series_id.up.sql:3]: Yeah, good call.

File: migrations/frontend/1528395784_insights_series_id.down.sql (hunk 1 above)
Comment (efritz) [reply migrations/frontend/1528395784_insights_series_id.down.sql:2]: I guess the up migration is idempotent, nvm then

**Review by efritz — APPROVED** [review]

**Pull request merged**